build_cpp.sh: remove duplicate -DCMAKE_BUILD_TYPE and be more verbose while compiling

meyerj opened the pull request.

meyerj committed:
build_cpp.sh: remove duplicate -DCMAKE_BUILD_TYPE and be more verbose while compiling

jubeira commented on build_cpp.sh:
```diff
@@ -108,7 +107,7 @@ catkin config \
     -DBUILD_TESTING=OFF -DCATKIN_ENABLE_TESTING=OFF
 
-catkin build
+catkin build --force-cmake --summary -ip1
```
This script has an unused `VERBOSE` flag above. Can we use it to set `-ip1`?

meyerj replied:
Done: https://github.com/Intermodalics/ros_android/pull/69/commits/d07817b30ed342443a203319fce634e084c38e02

jubeira reviewed (approved):
Just a minor comment, otherwise LGTM!

meyerj committed:
build_cpp: forward verbose flags to catkin

meyerj committed:
build_cpp.sh: remove catkin variable -DUSE_CATKIN

Not sure why it was needed. It is unused in the current catkin_ws.

meyerj committed:
build_cpp.sh: fix for standalone usage and allow to pass additional parameters to catkin

This is useful to rebuild only a single package with or without its dependencies.

CMAKE_PREFIX_PATH and CMAKE_INSTALL_PREFIX do not need to be set as they will be added by catkin_tools
depending on the workspace configuration (catkin config).

meyerj committed:
Merge remote-tracking branch 'origin/kinetic' into fix/duplicate-cmake-build-type

meyerj committed:
docker: clear entrypoint to not pollute the environment with stuff in /opt/ros/kinetic

meyerj commented:
I added some more extensions useful for debugging, e.g. to pass custom catkin arguments to `build_cpp.sh`.

- `CMAKE_PREFIX_PATH` and `CMAKE_INSTALL_PREFIX` cmake variables do not need to be set explicitly as they will be added by catkin_tools depending on the workspace configuration (`catkin config`).
- `USE_CATKIN` was also unused - at least `grep USE_CATKIN catkin_ws/src -R` does not return any result.
- Removed `ENTRYPOINT /ros_entrypoint.sh` defined in the docker base image such that the environment is clean and the build does not accidentally pick up the CMAKE_PREFIX_PATH set in the shell, e.g. when calling `build_cpp.sh` directly.

meyerj commented:
@jubeira Still good to merge? Some significant changes have been added since your [last review](https://github.com/Intermodalics/ros_android/pull/69#pullrequestreview-184158295).

jubeira commented:
Yes, I reviewed them yesterday and left it building from scratch, but I had to leave before it finished. I had no issues and the new changes are looking good; merging now!
